### PR TITLE
Removed single use of session holder and decouple drop in implementation

### PR DESCRIPTION
**Pull request by Robert-SD**

### CODE REVIEW

File: android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt:1]: Generated code - no review required.

File: ios/Classes/PlatformApi.swift
Comment (Robert-SD) [inline ios/Classes/PlatformApi.swift:1]: Generated code - no review required.

File: lib/src/generated/platform_api.g.dart
Comment (Robert-SD) [inline lib/src/generated/platform_api.g.dart:1]: Generated code - no review required.

File: android/src/main/kotlin/com/adyen/checkout/flutter/dropIn/DropInPlatformApi.kt [hunk 1]
```diff
@@ -50,7 +51,10 @@ class DropInPlatformApi(
     lateinit var dropInSessionLauncher: ActivityResultLauncher<SessionDropInResultContractParams>
     lateinit var dropInAdvancedFlowLauncher: ActivityResultLauncher<DropInResultContractParams>
 
-    override fun showDropInSession(dropInConfigurationDTO: DropInConfigurationDTO) {
+    override fun showDropInSession(
```
Comment (Robert-SD) [inline android/src/main/kotlin/com/adyen/checkout/flutter/dropIn/DropInPlatformApi.kt:54]: I will align the Android adjustment in another PR.

File: android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt [hunk 2]
```diff
@@ -1536,23 +1541,28 @@ private object DropInPlatformInterfaceCodec : StandardMessageCodec() {
...
 interface DropInPlatformInterface {
-  fun showDropInSession(dropInConfigurationDTO: DropInConfigurationDTO)
+  fun showDropInSession(dropInConfigurationDTO: DropInConfigurationDTO, sessionDTO: SessionDTO)
   fun showDropInAdvanced(dropInConfigurationDTO: DropInConfigurationDTO, paymentMethodsResponse: String)
   fun onPaymentsResult(paymentsResult: PaymentEventDTO)
   fun onPaymentsDetailsResult(paymentsDetailsResult: PaymentEventDTO)
   fun onDeleteStoredPaymentMethodResult(deleteStoredPaymentMethodResultDTO: DeletedStoredPaymentMethodResultDTO)
-  fun cleanUpDropIn()
+  fun cleanUpDropInAdvanced()
+  fun cleanUpDropInSession()
```
Comment (descorp) [inline android/src/main/kotlin/com/adyen/checkout/flutter/PlatformApi.kt:1566]: Do we need 2 separate methods?

**Pull request closed without merging**